static/pane-renaming.ts

```diff
@@ -25,32 +25,55 @@
 import $ from 'jquery';
 import _ from 'underscore';
 import { Tab } from 'golden-layout';
+import { EventEmitter } from 'events';
 
-//const Alert = require('alert').Alert;
+const Alert = require('alert').Alert;
 
-export class PaneRenaming {
-    public static registerCallback(pane: any): void {
-        // const alertSystem = new Alert();
-        // const addRenameButton = function (parentTab: Tab) {
-        //     return PaneRenaming.addRenameButton.call(this, parentTab, pane, alertSystem);
-        // };
-        // pane.container.on('tab', addRenameButton);
+export class PaneRenaming extends EventEmitter.EventEmitter{
+    private pane: any;
+    private alertSystem: any;
+    private state: any;
+
+    constructor(pane: any, state: any) {
+        super();
+        this.pane = pane;
+        this.alertSystem = new Alert();
```

Comment:
What you can do here about the alerty situation is to use the alert coming from the parent Pane (In most panes, it's the `this.alertSystem variable`) So something like this

```suggestion
    constructor(pane: any, state: any, alertSystem?: Alert) {
        super();
        this.pane = pane;
        if (alertSystem === undefined) {
            this.alertSystem = new Alert();
        } else {
            this.alertSystem = alertSystem;
        }
```